invenio_vocabularies/records/pidprovider.py

```diff
@@ -102,6 +102,27 @@ def create(cls, object_type=None, object_uuid=None, record=None, **kwargs):
         )
 
 
+class FromFieldProvider:
+    """PID from a metadata field provider.
+
+    This provider extracts the PID value from a field inside the
+    record metadata.
+    """
+
+    # FIXME: Provider.create usually is a class methods
+    #        but scheme is more suitable for an instance attr
+    field = "identifiers"
+    scheme = "orcid"
+
+    @classmethod
+    def create(cls, record):
+        """Extract the PID value."""
+        for id_ in record.get(cls.field):  # TODO: implement recursion?
+            if id_["scheme"] == cls.scheme:
+                return id_["identifier"]
+        return None
```

Comment:
Doubt: I'm a bit hesitant regarding this implicit "coupling" between the `identifiers` (or any) metadata field, and the systemfield itself. E.g. imagine one creates a record without including in `identifiers` a value with the required scheme, what happens to the `record.pid` value? I guess it's `None` which is an issue. 

In theory, the validation would need to happen on the service schema either to validate either:

- that an `identifiers` entry exists with `scheme: orcid`
- a `pid` value has been passed (and possibly validate that the `scheme` is correct?)

Reply:
Yes, this is the drawback. I think we can do it in two steps:
1. _pre load_: if there is no pid, and take it from a configured field `identifiers`
2. _validate_: if no pid (extracted before), then fail (`pid = ...(required=True)`)